### ISSUE TITLE
Autoencoder Training Loop

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/jubilant-palm-tree#36]
Description: Create a new training script (train_autoencoder.py). This script will be similar to the previous training script but adapted for the autoencoder.

Key Differences:

    It will only train the weights of the decoder. The encoder's weights will be frozen.

    It will use the new AST reconstruction loss function.

    The input and the target for the model are the same: the AST graph.

Definition of Done:

    The script can successfully train the autoencoder for at least two epochs.

    Training and validation loss decrease over time.

    The best decoder weights are saved to a file (e.g., best_decoder.pt).

After that, update the README to reflect the changes.
